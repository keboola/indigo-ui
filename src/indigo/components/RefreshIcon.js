import React from 'react';
import PropTypes from 'prop-types';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faSync } from '@fortawesome/free-solid-svg-icons/faSync';
import Loader from './Loader';

class RefreshIcon extends React.Component {
  render() {
    const { isLoading, title, ...remaining } = this.props;

    return (
      <span title={title}>
        {isLoading ? (
          <Loader />
        ) : (
          <FontAwesomeIcon icon={faSync} fixedWidth className="icon-clickable" {...remaining} />
        )}
      </span>
    );
  }
}

RefreshIcon.propTypes = {
  isLoading: PropTypes.bool.isRequired,
  title: PropTypes.string,
};

RefreshIcon.defaultProps = {
  title: 'Refresh',
};

export default RefreshIcon;
