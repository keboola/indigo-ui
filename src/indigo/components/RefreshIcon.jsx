import React from 'react';
import Loader from './Loader';

const RefreshIcon = React.createClass({
  propTypes: {
    isLoading: React.PropTypes.bool.isRequired,
    title: React.PropTypes.string
  },

  getDefaultProps() {
    return {
      title: 'Refresh'
    };
  },

  render() {
    const { isLoading, title, ...remaining } = this.props;
    return (
      <span title={title}>
        {isLoading
          ? <Loader/>
          : <span {...remaining} className="kbc-refresh kbc-icon-cw" />}
      </span>
    );
  },
});

export default RefreshIcon;
