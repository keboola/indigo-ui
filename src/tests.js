import renderer from 'react-test-renderer';

export const snapshot = (component) => {
  const tree = renderer
    .create(component)
    .toJSON();
  expect(tree).toMatchSnapshot();
};
