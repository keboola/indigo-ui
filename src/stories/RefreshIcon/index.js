import React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';

import RefreshIcon from '../../indigo/components/RefreshIcon';
import { boolean } from '@storybook/addon-knobs/dist/index';

storiesOf('RefreshIcon', module).add(
  'RefreshIcon',
  withInfo({
    text: `
        Refresh icon with loader.
        `,
    inline: true,
  })(() => {
    const isLoading = boolean('isLoading', false);
    return <RefreshIcon isLoading={isLoading} />;
  })
);
