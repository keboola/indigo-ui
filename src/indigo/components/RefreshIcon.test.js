import React from 'react';

import { matchesSnapshot } from '../../tests';
import RefreshIcon from './RefreshIcon';

describe('<RefreshIcon />', () => {
  it('RefreshIcon - loading', () => {
    matchesSnapshot(<RefreshIcon isLoading />);
  });

  it('RefreshIcon - not loading', () => {
    matchesSnapshot(<RefreshIcon isLoading={false} />);
  });

  it('RefreshIcon - loading with custom title', () => {
    matchesSnapshot(<RefreshIcon isLoading title="Click to reload" />);
  });
});
