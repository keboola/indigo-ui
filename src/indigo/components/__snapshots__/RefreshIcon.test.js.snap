// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RefreshIcon /> RefreshIcon - loading 1`] = `
<span
  title="Refresh"
>
  <span
    className="fa fa-spin fa-spinner"
  />
</span>
`;

exports[`<RefreshIcon /> RefreshIcon - loading with custom title 1`] = `
<span
  title="Click to reload"
>
  <span
    className="fa fa-spin fa-spinner"
  />
</span>
`;

exports[`<RefreshIcon /> RefreshIcon - not loading 1`] = `
<span
  title="Refresh"
>
  <span
    className="kbc-refresh kbc-icon-cw"
  />
</span>
`;
